extensions/amp-link-rewriter/0.1/test/test-noscope-document.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {LinkRewriter} from '../link-rewriter';
+import helpersMaker from './test-helpers';
+
+describes.fakeWin(
```

Comment:
Is there any way we can split up these `it()` blocks and move some setup logic to a `beforeEach` block instead? The tests are pretty long and look like they all share common setup.

Reply:
I think I could make the helper function for part of the code similar to this (from 188 to 215)
https://github.com/ampproject/amphtml/pull/33754/files#diff-e027941a001346701616405926e26ede898ae4833804ee00368957d5ce8e8a85R188

Reply:
I'll leave it up to you :)

Reply:
https://github.com/ampproject/amphtml/pull/33754/commits/d00e7e5e759708739b424035147be5eab2682764#diff-e027941a001346701616405926e26ede898ae4833804ee00368957d5ce8e8a85R111